examples/gno.land/p/demo/paginate/paginate_test.gno

```diff
@@ -0,0 +1,117 @@
+package paginate
+
+import (
+	"strconv"
+	"testing"
+
+	"gno.land/p/demo/avl"
+)
+
+func paginTest(page int, size int, reverse bool) string {
+	if reverse {
+		return "://here is the url/" + strconv.Itoa(page)
+	}
+	return "://here is the url/" + strconv.Itoa(size-page+1)
+}
+
+type testItems struct {
+	content string
+}
+
+func (t *testItems) Render() string {
+	return "CONTENT OF TESTITEM: { " + t.content + " }"
+}
+
+var mock avl.Tree
+
+func init() {
+	mock.Set("1", &testItems{content: "content 1"})
+	mock.Set("2", &testItems{content: "content 2"})
+	mock.Set("3", &testItems{content: "content 3"})
+	mock.Set("4", &testItems{content: "content 4"})
+	mock.Set("5", &testItems{content: "content 5"})
+	mock.Set("6", &testItems{content: "content 6"})
+	mock.Set("7", &testItems{content: "content 7"})
+	mock.Set("8", &testItems{content: "content 8"})
+}
+
+func TestPaginate(t *testing.T) {
+	const testString = `[1](%3A//here%20is%20the%20url/1) *2* [3](%3A//here%20is%20the%20url/3) ... [6](%3A//here%20is%20the%20url/6) [7](%3A//here%20is%20the%20url/7) [8](%3A//here%20is%20the%20url/8) 
+
+CONTENT OF TESTITEM: { content 2 }
+
+[1](%3A//here%20is%20the%20url/1) *2* [3](%3A//here%20is%20the%20url/3) ... [6](%3A//here%20is%20the%20url/6) [7](%3A//here%20is%20the%20url/7) [8](%3A//here%20is%20the%20url/8) 
+`
+	res := RenderTreeByPage(mock, paginTest, 2, 1, true)
```

Comment:
Can you add 2 more cases in the tests, related to the size parameter:
- size > 1 and <= `mock.Size()`
- size > `mock.Size()`

Reply:
done for the first test but I don't see any way to test a panic so the second is impossible to test